lib/tensorflow/tensor.rb

```diff
@@ -182,17 +182,39 @@ def ruby_array_to_c(array, type)
   #   - Value of the element contained in the specified position in the tensor.
   #
   def getval(dimension)
-    raise("Invalid dimension array passed as input.",ShapeError) if dimension.length != dimensions.length
+    raise("Invalid dimension array passed as input.",ShapeError) if dimension.length != shape.length
     (0..dimension.length-1).each do |i|
-      raise("Invalid dimension array passed as input.",ShapeError) if dimension[i] > dimensions[i] || dimension[i] < 1 || !(dimension[i].is_a? Integer)
+      raise("Invalid dimension array passed as input.",ShapeError) if dimension[i] > shape[i] || dimension[i] < 1 || !(dimension[i].is_a? Integer)
     end
-    sum = dimension[dimension.length - 1]  - 1
-    prod = dimensions[dimensions.length - 1]
+    sum = dimension.last - 1
+    prod = shape.last
     (0..dimension.length - 2).each do |i|
        sum += (dimension[dimension.length - 2 - i] - 1) * prod
-       prod *= dimensions[dimensions.length - 2 - i]
+       prod *= shape[shape.length - 2 - i]
     end
 
     flatten[sum]
   end
+
+  private
+
+  #
+  # Recursively finds the shape of the input array.
+  #
+  # * *Returns* :
+  #   - Dimension array `[[2], [4]].shape` => `[2, 1]`
+  #
+  # TODO: Handle non-rectangular arrays and raise error if mixed data types
+  def self.set_shape(value)
```

Comment:
I'm not sure that close following of Python conventions is the best way to design Ruby library. I've been in those discussions several times (for example, discussing some of new/existing features of [daru](https://github.com/v0dro/daru)), and whether they should look like genuine Ruby or "something familiar for similar Python library users" (pandas for daru), and the former option (make it good Ruby, not "pythonic dialect of Ruby") always wins.

In other words, if there would be "method to set shape", it should be called `shape=` in Ruby, not `set_shape` in any circumstances. (And, as far as I can see, Python's `set_shape` doesn't do analysis of incoming object's shape.)